include/wx/charts/wxmath2dplotoptions.h

```diff
@@ -74,6 +74,8 @@ class wxMath2DPlotOptions : public wxChartOptions
     const AxisFunc& GetAxisFuncY() const;
     void SetAxisFuncX(const AxisFunc &newfunc);
     void SetAxisFuncY(const AxisFunc &newfunc);
+    bool isNeedsFit() const;
```

Comment:
Having the options object know about the state of the chart is going to cause issues. It's going to be difficult to keep updating m_needsFit so that it is in the right state.
Although it is going to add to the number of classes it would be best to use the observer pattern for this. Then the chart can register itself as an observer with the options and get notified of relevant changes and there is no need for the options to keep track of the state.

Reply:
I think I just overcomplicated this, so there not needed observer or even check for changes. If options changed usually that's mean that need relayout of chart because case when new options exactly equal old options very rarely and no need extra checks.  So I avoid it in my new update.